core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java

```diff
@@ -81,6 +82,11 @@ public static TransmissionNetworkOutput create() {
     }
 
     public static TransmissionNetworkOutput create(String endpoint) {
+        if (!ClassDataUtils.INSTANCE.isClassExists("org.apache.http.conn.HttpClientConnectionManager")) {
+            String errorMessage = String.format("HttpClient Jars mismatch: please make sure version 4.3.1 of HttpClient is used");
+            throw new IllegalStateException(errorMessage);
```

Comment:
According to the description of [IllegalStateException](http://docs.oracle.com/javase/7/docs/api/java/lang/IllegalStateException.html):

> Signals that a method has been invoked at an illegal or inappropriate time

So I'm not sure that this is the right exception to throw here.
IMHO more suitable exceptions would be ClassNotFound exception with the message above.